runtime/js-compute-runtime/builtins/cache-simple.cpp

```diff
@@ -213,6 +212,12 @@ JS::Result<std::string> createSurrogateKeysFromCacheKey(JSContext *cx, std::stri
   return JS::Result<std::string>(surrogate_keys);
 }
 
+#define CANCEL_TRANSACTION(cx, promise, handle)                                                    \
+  if (auto *err = handle.transaction_cancel().to_err()) {                                          \
+    host_api::handle_fastly_error(cx, *err, __LINE__, __func__);                                   \
+  }                                                                                                \
+  return RejectPromiseWithPendingError(cx, promise);
+
```

Comment:
It would add some code back in at the call-site, but what do you think about removing the return here and adding it back after each call to `CANCEL_TRANSACTION`? Using `return` in macros makes it hard to tell locally that control flow won't continue after the macro call.